File: frontend/src/api/locations.ts
import api from './client'
import { staticLocations } from './staticClient'

export interface Location {
  name: string
  address: string
  city: string
  state: string
  zip: string
  country: string
  hours: string | null
  phone: string | null
  full_address: string
  maps_embed_url_keyed: string
  maps_embed_url_legacy: string
  maps_link_url: string
}

// ── Toggle: set VITE_STATIC_MODE=true to bypass the backend (GitHub Pages) ──
const STATIC = import.meta.env.VITE_STATIC_MODE === 'true'

export const getLocations = (): Promise<Location[]> => {
  if (STATIC) return staticLocations<Location[]>()
  return api.get<Location[]>('/locations/').then(r => r.data)
}

File: backend/app/routers/locations.py
from fastapi import APIRouter
from pydantic import BaseModel
from typing import Optional, List
from ..services.locations_loader import load_locations
from ..config import settings

router = APIRouter(prefix="/api/locations", tags=["locations"])


class LocationOut(BaseModel):
    name: str
    address: str
    city: str
    state: str
    zip: str
    country: str
    hours: Optional[str]
    phone: Optional[str]
    full_address: str
    maps_embed_url_keyed: str
    maps_embed_url_legacy: str
    maps_link_url: str


@router.get("/", response_model=List[LocationOut])
def list_locations():
    locs = load_locations(
        settings.STORE_SLUG,
        s3_bucket=settings.MENU_S3_BUCKET,
        aws_region=settings.AWS_REGION,
        api_key=settings.GOOGLE_MAPS_API_KEY,
    )
    return [
        LocationOut(
            name=l.name,
            address=l.address,
            city=l.city,
            state=l.state,
            zip=l.zip,
            country=l.country,
            hours=l.hours,
            phone=l.phone,
            full_address=l.full_address,
            maps_embed_url_keyed=l.maps_embed_url_keyed(),
            maps_embed_url_legacy=l.maps_embed_url_legacy,
            maps_link_url=l.maps_link_url,
        )
        for l in locs
    ]
